python/pyspark/sql/tests/test_udtf.py

```diff
@@ -247,7 +247,8 @@ class TestUDTF:
             def eval(self, a: int):
                 yield
 
-        assertDataFrameEqual(TestUDTF(lit(1)), [Row(a=None)])
+        if have_pyarrow:
+            assertDataFrameEqual(TestUDTF(lit(1)), [Row(a=None)])
```

Comment:
@allisonwang-db why do we need Arrow here?

Reply:
It's not UDTF, it's this `assertDataFrameEqual` requires Arrow: https://github.com/apache/spark/pull/44778/files#r1467378305
This should be fixed.